app/src/main/java/com/infomaniak/drive/utils/MediaFoldersProvider.kt

```diff
@@ -149,11 +149,20 @@ object MediaFoldersProvider {
         coroutineScope?.ensureActive()
         if (path.startsWith("Android/media/${BuildConfig.APPLICATION_ID}")) return null
         return MediaFolder.findById(realm, folderId)?.let { mediaFolder ->
-            mediaFolder.apply { if (mediaFolder.name != folderName) mediaFolder.storeOrUpdate() }
+            mediaFolder.apply {
+                if (mediaFolder.name != folderName || mediaFolder.path != path) {
+                    mediaFolder.storeOrUpdate(path)
+                }
+            }
         } ?: let {
             val isFirstConfiguration = !AccountUtils.isEnableAppSync()
             val isSynced = isFirstConfiguration && path.contains("${Environment.DIRECTORY_DCIM}/")
-            MediaFolder(folderId, folderName, isSynced).apply { storeOrUpdate() }
+            MediaFolder(
+                id = folderId,
+                name = folderName,
+                isSynced = isSynced,
+                path = path,
+            ).apply { storeOrUpdate() }
```

Comment:
```suggestion
            ).also(::storeOrUpdate)
 ```

Reply:
Maybe even put it on one line, the argument name aren't really useful here

Reply:
I have to do `MediaFolder::storeOrUpdate`.